project4/proto/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+include(FindProtobuf)
+find_package(Protobuf REQUIRED)
+
+include_directories(${PROTOBUF_INCLUDE_DIR})
+PROTOBUF_GENERATE_CPP(PROTO_SRC PROTO_HEADER pub_sub_messages.proto)
+add_library(pub_sub_proto ${PROTO_HEADER} ${PROTO_SRC})
```

Comment:
You will need a different name for this target. I'm actually surprised this builds as it is.